src/java.desktop/share/classes/java/awt/font/TextJustifier.java

```diff
@@ -155,7 +155,10 @@
             boolean absorbing = hitLimit && absorbweight > 0;
 
             // predivide delta by weight
-            float weightedDelta = delta / weight; // not used if weight == 0
+            float weightedDelta = 0;
+            if (weight != 0) { // not used if weight == 0
```

Comment:
Can it ever be -ve? Maybe we can do weight > 0 check just as we do for absorbweight?

Reply:
Hi, I am not sure about the  weight > 0  check ; weight is initialized with 0:   weight = 0;   and later some values are potentially added up to weight:   weight += gi.weight;
I am not sure about those gi.weight values, maybe they can be negative too ?

Reply:
Nothing throws an exception or otherwise prevent this being negative but that would be a weird usage. Typically the weight is either zero or based on the font size .. which ought not to be negative but I don't think anything prevents it and I think we would treat it essentially as a transform. So If you really want to be careful here, then yes assume weight could be negative.

Reply:
By that same logic, then shouldn't absorbweight also be checked as != 0 instead of > 0 as that also uses += gi.weight

Reply:
Hi, I am not sure about the absorbweight  check; but  currently the calculated value weightedAbsorb is only used when absorbing is true. And  there the > 0 check is present too

boolean absorbing = hitLimit && absorbweight > 0;

Reply:
I meant that since we are dividing by weight and absorbweight 
` weightedDelta = delta / weight;`
and
 weightedAbsorb = (delta - gslimit) / absorbweight;
where both weight and absorbweight uses += gi.weight values so if one is checking for >0 
then to be consistent, in my opinion, 
other one should be same or absorbweight also probably should check !=0.

Reply:
should we then better check  for 
 if (weight > 0) 
instead of
 if (weight != 0) 
like it is done for absorbweight ?
(If you think consistency is important here)

Reply:
I guess since it was mentioned before that nothing prevents the value from being -ve,
I guess better will be to use != 0 check for absorbweight too?

Reply:
okay, I adjusted the check for absorbweight .